Source/controls/plrctrls.cpp

```diff
@@ -445,12 +462,32 @@ void Interact()
 		NetSendCmdLocParam1(true, CMD_TALKXY, towners[pcursmonst].position, pcursmonst);
 	} else if (pcursmonst != -1) {
 		if (plr[myplr]._pwtype != WT_RANGED || CanTalkToMonst(pcursmonst)) {

```

Comment:
Not sure about this `CanTalkToMonst(pcursmonst)` change, perhaps we should still walk to the monster?

Reply:
My reasoning here is based on games that include controller support (including D3 and D2R).

On these games the user controls the character more directly (as if they are the character), and not just command them to do things (which is what happens when using a mouse on these games).